api/v1/models.go

```diff
@@ -316,3 +319,24 @@ func pageNumberToToken(page database.VulnerabilityNotificationPageNumber, key st
 
 	return string(tokenBytes)
 }
+
+func tokenToNumber(token, key string) (int, error) {
```

Comment:
We could definitely only have one function for marshaling and one function for unmarshaling for tokens. Just make it only return an error and you can pass a newly allocated type by reference similar to [normal JSON unmarshalling](https://godoc.org/encoding/json#Unmarshal).